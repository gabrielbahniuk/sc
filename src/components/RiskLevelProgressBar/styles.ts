import { appConfig } from '../../config'
import styled from 'styled-components'

const { MAXIMUM_RISK_LEVEL } = appConfig

export const ProgressBar = styled.div`
  background: #333;
  border-radius: 13px;
  height: 1.5rem;
  width: 100%;
  padding: 3px;
  transition: width 0.2s;
  margin-top: 1rem;

  span {
    content: '';
    display: block;
    background-color: ${({ riskLevel }): string =>
      riskLevel <= 8 ? 'var(--ok-color)' : riskLevel <= 16 ? 'var(--alert-color)' : 'var(--danger-color)'};
    width: ${({ riskLevel }): number => (riskLevel / MAXIMUM_RISK_LEVEL) * 100}%;
    height: 100%;
    border-radius: 9px;
    transition: width 0.3s;
  }
`

export const ProgressLabelContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  span {
    color: var(--text-color);
  }
`
