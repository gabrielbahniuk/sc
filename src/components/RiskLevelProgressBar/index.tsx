import React from 'react'

import { ProgressBar, ProgressLabelContainer } from './styles'

type RiskLevelProgressBar = {
  riskLevel: number
}

const RiskLevelProgressBar: React.FC<RiskLevelProgressBar> = ({ riskLevel }) => {
  return (
    <>
      <ProgressBar riskLevel={riskLevel}>
        <span data-cy="progressBar"></span>
      </ProgressBar>
      <ProgressLabelContainer>
        <span>Low Risk</span>
        <span>High Risk</span>
      </ProgressLabelContainer>
    </>
  )
}

export default RiskLevelProgressBar
