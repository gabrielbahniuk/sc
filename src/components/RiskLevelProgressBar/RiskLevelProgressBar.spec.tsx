import React from 'react'
import RiskLevelProgressBar from '.'
import { create } from 'react-test-renderer'

describe('Test Progress Bar', () => {
  it('should render a progress bar', () => {
    const progressBar = create(<RiskLevelProgressBar riskLevel={3} />).toJSON()
    expect(progressBar).toMatchSnapshot()
  })
})
