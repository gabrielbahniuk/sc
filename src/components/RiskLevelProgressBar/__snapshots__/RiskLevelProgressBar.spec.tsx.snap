// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Progress Bar should render a progress bar 1`] = `
Array [
  <div
    className="sc-bdnxRM WVzIi"
  >
    <span
      data-cy="progressBar"
    />
  </div>,
  <div
    className="sc-gtsrHT htXvXo"
  >
    <span>
      Low Risk
    </span>
    <span>
      High Risk
    </span>
  </div>,
]
`;
